### PR TITLE
Update to angular front-end to use with backend middleware libraries

### PR DESCRIPTION
**Pull request by vijetmahabaleshwar-okta**
Updated the following 

1. Removed using AuthJS to redirect to login/logout routes
2. Updated the mustache templates to use generic information
3. Bumped up the version in package.json

To test this, you can use the express-js backend sample here - https://github.com/okta/samples-nodejs-express-4/pull/26

In the backend app's package.json, you can refer to this module to see the e2e flow for auth-code exchange.

### CODE REVIEW

File: tools/templates/profile.mustache [hunk 1]
```diff
@@ -55,27 +51,12 @@
           </a>
         </div>
       </li>
+
       <li>
-        Exchange the <strong>authorization code</strong> for an
```
Comment (robertjd) [inline tools/templates/profile.mustache]: I think this information and the link to DOC is useful and should be brought back, but it should be mentioned that the back-end is handling this for you.

File: tools/templates/overview.mustache [hunk 2]
```diff
@@ -203,16 +203,7 @@
        clientId: '{{config.oidc.clientId}}',
```
Comment (robertjd) [inline tools/templates/overview.mustache]: In this overview template, remove the login-custom link (for now).  We'll bring this back when we revamp the samples to the V2 strategy.

File: tools/templates/profile.mustache (hunk 1 above)
Comment (vijetmahabaleshwar-okta) [reply tools/templates/profile.mustache]: Done

File: tools/templates/overview.mustache (hunk 2 above)
Comment (vijetmahabaleshwar-okta) [reply tools/templates/overview.mustache]: Done

**Comment by jmelberg-okta** [comment]
@vijetmahabaleshwar-okta We also include a local copy of the `samples-nodejs-express-4` app. See [tools/dev-server](https://github.com/okta/samples-js-angular-1/tree/master/tools/dev-server). Does it make sense to include the modified sample there too?

**Comment by robertjd** [comment]
Per the last two comments, let's just rev the major version on this, and add a note to the readme that this is now an upgraded sample and that folks should chat with us before trying to use it in a new sample project, especially because we are now going to refactor our samples per the v2 plan.

**Comment by vijetmahabaleshwar-okta** [comment]
@jmelberg-okta, @robertdamphousse-okta 
Will update README saying this version is for server 2.x samples only.
I think it'll be nice to update the express server in [https://github.com/okta/samples-js-angular-1/tree/master/tools/dev-server](url). Let me do that as well.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,17 @@
 Angular 1 Samples
 ======
 
+## Note
+This branch of Angular sample is not compatible with our back end samples listed in [Using a different back end](#using-a-different-back-end)
+To use Angular sample with these back end samples, please checkout [Version 1.10](#https://github.com/okta/samples-js-angular-1/tree/1.10)
```
Comment (jmelberg-okta) [inline README.md]: Lets add an installation script here:
```bash
npm install @okta/samples-js-angular-1@1.10.0
```

File: package.json [hunk 4]
```diff
@@ -36,6 +37,8 @@
     "babel-polyfill": "6.16.0",
     "babel-preset-es2015": "6.18.0",
     "copy-webpack-plugin": "4.0.0",
+    "express": "4.14.0",
+    "express-session": "^1.15.5",
```
Comment (jmelberg-okta) [inline package.json]: Lets lock down this version as we do other dependencies: `1.15.5`

File: README.md (hunk 3 above)
Comment (jmelberg-okta) [reply README.md]: Also, we should split this into its own section. I'd probably fill this `README` with all info related to `2.x`, then link to a deprecated `README` file with instructions on running `1.x`

Comment (vijetmahabaleshwar-okta) [reply README.md]: Agreed. 
Have updated the README to include stuff only relevant to 2.x
Added a link to deprecated README with instructions running 1.x

File: package.json (hunk 4 above)
Comment (vijetmahabaleshwar-okta) [reply package.json]: Done

File: README.md [hunk 5]
```diff
@@ -59,34 +59,44 @@ To run this sample:
     ```
 
 5. You're all set! Open a browser window to [http://localhost:3000](http://localhost:3000).
+#### Note
```
Comment (jmelberg-okta) [inline README.md]: nit: Replace `####` with `>`:

> Note: Use the Chrome browser if you're running the mock-okta server.

File: README.md [hunk 6]
```diff
@@ -59,34 +59,44 @@ To run this sample:
...
+Open a new terminal window, and run the Express server.
+   
+   ```bash
+   # Starts the sample back end on http://localhost:3000
+   [samples-js-angular-1]$ npm start
+   ```
+   
+You're all set! Open a browser window to [http://localhost:3000](http://localhost:3000).
 
-## Using a different back end
```
Comment (jmelberg-okta) [inline README.md]: We still need to keep this. Can you create a new `README-v1.md` file that contains the "old README"? Then we can link to it in our callout above.

Comment (vijetmahabaleshwar-okta) [reply README.md]: I have linked the older README at the top, where I ask users to use previous version if they want to use different back-ends

File: README.md (hunk 5 above)
Comment (vijetmahabaleshwar-okta) [reply README.md]: Done

**Pull request merged**